### PR TITLE
[ie/twitter:spaces] Fix format protocol

### LINKED ISSUE #7536: [twitter:spaces] Invalid data found when processing input
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I'm running yt-dlp version **2023.07.06** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Japan

### Provide a description that is worded well enough to be understood

When trying to download a twitter spaces audio, I encountered the error "Invalid data found when processing input".
I tried several times, but got this same error.
I was running yt-dlp version 2023.07.06.194006.
I would be grateful if someone would fix this issue.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', '-f', 'ba', '-o', '%(uploader_id)s_%(upload_date)s_%(title)s_%(id)s.%(ext)s', '-P', '.', '--netrc', 'https://twitter.com/i/spaces/1eaKbrQbjoRKX']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version nightly@2023.07.06.194006 [92315c037] (zip)
[debug] Python 3.10.4 (CPython arm64 64bit) - macOS-13.4-arm64-arm-64bit (OpenSSL 1.1.1u  30 May 2023)
[debug] exe versions: ffmpeg 6.0 (setts), ffprobe 6.0
[debug] Optional libraries: certifi-2021.10.08, sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1855 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-nightly-builds/releases/latest
Available version: nightly@2023.07.06.194006, Current version: nightly@2023.07.06.194006
Current Build Hash: 938b91103cd7dc7544ea4eadd44d8488ec66e0ab237d0aee211baa5244b16fb9
yt-dlp is up to date (nightly@2023.07.06.194006)
[twitter:spaces] Downloading login page
[twitter:spaces] Downloading flow token
[twitter:spaces] Submitting JS instrumentation response
[twitter:spaces] Submitting username
[twitter:spaces] Submitting password
[twitter:spaces] Submitting account duplication check
[twitter:spaces] Logging in
[twitter:spaces] Extracting URL: https://twitter.com/i/spaces/1eaKbrQbjoRKX
[twitter:spaces] 1eaKbrQbjoRKX: Downloading GraphQL JSON
[twitter:spaces] 28_1664225270709256192: Downloading legacy API JSON
[twitter:spaces] 28_1664225270709256192: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[info] 1eaKbrQbjoRKX: Downloading 1 format(s): 0
[debug] Invoking hlsnative downloader on "https://prod-fastly-ap-northeast-1.video.pscp.tv/Transcoding/v1/hls/uoJ7s877sh5vLzDFNKzftjMMb1_Jd7m5i-ePKC63ZK0LnMLjztwQQCNGMZu7WHDdBYkKHM7slVLUMtGLkQhxvw/non_transcode/ap-northeast-1/periscope-replay-direct-prod-ap-northeast-1-public/audio-space/playlist_16761126492889887124.m3u8?type=replay"
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 125
[download] Destination: ./tomeru_ikinone_20230601_あ_1eaKbrQbjoRKX.m4a
[download] 100% of    4.46MiB in 00:00:06 at 705.21KiB/s
[FixupM4a] Correcting container of "./tomeru_ikinone_20230601_あ_1eaKbrQbjoRKX.m4a"
[debug] ffmpeg command line: ffmpeg -y -loglevel repeat+info -i 'file:./tomeru_ikinone_20230601_あ_1eaKbrQbjoRKX.m4a' -map 0 -dn -ignore_unknown -c copy -f mp4 -movflags +faststart 'file:./tomeru_ikinone_20230601_あ_1eaKbrQbjoRKX.temp.m4a'
[debug] ffmpeg version 6.0 Copyright (c) 2000-2023 the FFmpeg developers
  built with Apple clang version 14.0.3 (clang-1403.0.22.14.1)
  configuration: --prefix=/opt/homebrew/Cellar/ffmpeg/6.0 --enable-shared --enable-pthreads --enable-version3 --cc=clang --host-cflags= --host-ldflags= --enable-ffplay --enable-gnutls --enable-gpl --enable-libaom --enable-libaribb24 --enable-libbluray --enable-libdav1d --enable-libmp3lame --enable-libopus --enable-librav1e --enable-librist --enable-librubberband --enable-libsnappy --enable-libsrt --enable-libsvtav1 --enable-libtesseract --enable-libtheora --enable-libvidstab --enable-libvmaf --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libxvid --enable-lzma --enable-libfontconfig --enable-libfreetype --enable-frei0r --enable-libass --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libspeex --enable-libsoxr --enable-libzmq --enable-libzimg --disable-libjack --disable-indev=jack --enable-videotoolbox --enable-audiotoolbox --enable-neon
  libavutil      58.  2.100 / 58.  2.100
  libavcodec     60.  3.100 / 60.  3.100
  libavformat    60.  3.100 / 60.  3.100
  libavdevice    60.  1.100 / 60.  1.100
  libavfilter     9.  3.100 /  9.  3.100
  libswscale      7.  1.100 /  7.  1.100
  libswresample   4. 10.100 /  4. 10.100
  libpostproc    57.  1.100 / 57.  1.100
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x124633930] Format mov,mp4,m4a,3gp,3g2,mj2 detected only with low score of 1, misdetection possible!
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x124633930] moov atom not found
file:./tomeru_ikinone_20230601_あ_1eaKbrQbjoRKX.m4a: Invalid data found when processing input

ERROR: Postprocessing: file:./tomeru_ikinone_20230601_あ_1eaKbrQbjoRKX.m4a: Invalid data found when processing input
Traceback (most recent call last):
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3439, in process_info
    replace_info_dict(self.post_process(dl_filename, info_dict, files_to_move))
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3621, in post_process
    info = self.run_all_pps('post_process', info, additional_pps=info.get('__postprocessors'))
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3603, in run_all_pps
    info = self.run_pp(pp, info)
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3581, in run_pp
    files_to_delete, infodict = pp.run(infodict)
  File "/usr/local/bin/yt-dlp/yt_dlp/postprocessor/common.py", line 24, in run
    ret = func(self, info, *args, **kwargs)
  File "/usr/local/bin/yt-dlp/yt_dlp/postprocessor/common.py", line 129, in wrapper
    return func(self, info)
  File "/usr/local/bin/yt-dlp/yt_dlp/postprocessor/ffmpeg.py", line 882, in run
    self._fixup('Correcting container', info['filepath'], [*self.stream_copy_opts(), '-f', 'mp4'])
  File "/usr/local/bin/yt-dlp/yt_dlp/postprocessor/ffmpeg.py", line 863, in _fixup
    self.run_ffmpeg(filename, temp_filename, options)
  File "/usr/local/bin/yt-dlp/yt_dlp/postprocessor/ffmpeg.py", line 374, in run_ffmpeg
    return self.run_ffmpeg_multiple_files([path], out_path, opts, **kwargs)
  File "/usr/local/bin/yt-dlp/yt_dlp/postprocessor/ffmpeg.py", line 329, in run_ffmpeg_multiple_files
    return self.real_run_ffmpeg(
  File "/usr/local/bin/yt-dlp/yt_dlp/postprocessor/ffmpeg.py", line 367, in real_run_ffmpeg
    raise FFmpegPostProcessorError(stderr.strip().splitlines()[-1])
yt_dlp.postprocessor.ffmpeg.FFmpegPostProcessorError: file:./tomeru_ikinone_20230601_あ_1eaKbrQbjoRKX.m4a: Invalid data found when processing input
```

### PR DESCRIPTION
**Pull request by bashonly**
Reverts the `entry_protocol` for `_extract_m3u8_formats` to `m3u8`, so as to force ffmpeg as downloader for both live and replay Spaces.

Closes #7536

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
🐦🎙️🛠️

Add support for downloading Twitter Spaces and fix m3u8 extraction. Use `ffmpeg` to download m3u8 formats from `yt_dlp/extractor/twitter.py` to avoid missing segments.

> _`ffmpeg` needed_
> _to extract `m3u8` formats_
> _falling leaves of sound_

### Walkthrough
* Add support for downloading Twitter Spaces ([link](https://github.com/yt-dlp/yt-dlp/pull/7550/files?diff=unified&w=0#diff-8b22c3c3c6185164ce8c3e82541f91c728397dec66456eef7e05f664d5729103L1558-R1562),                       F0L1648R

</details>

### CODE REVIEW

**Review by pukkandan — APPROVED** [review]
I suggest either linking to the issue so we have an example URL

File: yt_dlp/extractor/twitter.py [hunk 1]
```diff
@@ -1555,13 +1555,11 @@ def _real_extract(self, url):
...
-            formats = self._extract_m3u8_formats(
-                source, metadata['media_key'], 'm4a', live=is_live, fatal=False,
-                headers={'Referer': 'https://twitter.com/'}) if source else []
+            formats = self._extract_m3u8_formats(  # XXX: Needs ffmpeg as downloader
+                source, metadata['media_key'], 'm4a', entry_protocol='m3u8', live=is_live,
+                headers={'Referer': 'https://twitter.com/'}, fatal=False) if source else []
             for fmt in formats:
                 fmt.update({'vcodec': 'none', 'acodec': 'aac'})
-                if not is_live:
-                    fmt['container'] = 'm4a_dash'
```
Comment (pukkandan) [inline yt_dlp/extractor/twitter.py:1564]: Add this anyway, since use can force native with `--downloader native`

Comment (bashonly) [reply yt_dlp/extractor/twitter.py]: It does still fixup when ffmpeg is used. Should we still add it back though? I have added it for now

Btw here's an example where native+fixup works (I wasn't just imagining it):
```
$ yt_dlp --cookies-from-browser firefox::twitter --downloader native 'https://twitter.com/i/spaces/1vAxRAVQWONJl'
[Cookies] Extracting cookies from firefox
[Cookies] Extracted 59 cookies from firefox
[twitter:spaces] Extracting URL: https://twitter.com/i/spaces/1vAxRAVQWONJl
[twitter:spaces] 1vAxRAVQWONJl: Downloading GraphQL JSON
[twitter:spaces] 28_1646577130061590528: Downloading legacy API JSON
[twitter:spaces] 28_1646577130061590528: Downloading m3u8 information
[info] 1vAxRAVQWONJl: Downloading 1 format(s): 0
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 829
[download] Destination: 20230413_googlecloud_1vAxRAVQWONJl_Framing_Up_FinOps_-_Billing_Tools.m4a
[download] 100% of   31.58MiB in 00:03:07 at 172.21KiB/s
[FixupM4a] Correcting container of "20230413_googlecloud_1vAxRAVQWONJl_Framing_Up_FinOps_-_Billing_Tools.m4a"
$
```

Comment (pukkandan) [reply yt_dlp/extractor/twitter.py]: > It does still fixup when ffmpeg is used.

```diff
diff --git a/yt_dlp/YoutubeDL.py b/yt_dlp/YoutubeDL.py
index d0be3c964..cc1741b75 100644
--- a/yt_dlp/YoutubeDL.py
+++ b/yt_dlp/YoutubeDL.py
@@ -3418,7 +3418,7 @@ def ffmpeg_fixup(cndn, msg, cls):
                     postprocessed_by_ffmpeg = info_dict.get('requested_formats') or any((
                         isinstance(pp, FFmpegVideoConvertorPP)
                         and resolve_recode_mapping(ext, pp.mapping)[0] not in (ext, None)
-                    ) for pp in self._pps['post_process'])
+                    ) for pp in self._pps['post_process']) or fd == FFmpegFD

                     if not postprocessed_by_ffmpeg:
                         ffmpeg_fixup(ext == 'm4a' and info_dict.get('container') == 'm4a_dash',
```

This should fix it

**Pull request merged**